src/ln/router.rs

```diff
@@ -426,6 +438,38 @@ impl RoutingMessageHandler for Router {
 
 		Ok(msg.contents.excess_data.is_empty())
 	}
+
+	fn get_next_announcements(&self, starting_point: u64, batch_amount: u8)->(Vec<msgs::ChannelAnnouncement>, Vec<msgs::ChannelUpdate>, Vec<msgs::NodeAnnouncement>, u64){
+		let mut channel_announcements = Vec::new();
+		let mut channel_updates = Vec::new();
+		let mut node_announcements = Vec::new();
+		let network = self.network_map.read().unwrap();
+		let mut starting_for_next = starting_point;
+		for x in 0..batch_amount{
+			if let Some(ref value) = network.channels.range(starting_for_next..).next(){
```

Comment:
This is ineffecient - you should get the iterator once then iterate through it instead of getting the iterator batch_amount times.